test/tests/functional/pbs_equiv_classes.py

```diff
@@ -390,6 +390,39 @@ def test_user_queue(self):
         self.scheduler.log_match("Number of job equivalence classes: 3",
                                  starttime=self.t)
 
+    def test_user_queue_without_limits(self):
+        """
+        Test that jobs from different users submitted to a queue without
+        a user limit set, will not create a multiple equivalence classes.
+        """
+
+        self.server.manager(MGR_CMD_SET, SERVER,
+                            {'scheduling': 'False'})
+
+        self.server.manager(MGR_CMD_SET, QUEUE,
+                            {'max_run': '[u:PBS_GENERIC=4]'}, id='workq')
+
+        # Eat up all the resources, this job will make first equiv class
+        a = {'Resource_List.select': '1:ncpus=8'}
+        J = Job(TEST_USER, attrs=a)
+        self.server.submit(J)
+
+        # Create a new queue and submit jobs to this queue
+        a = {'queue_type': 'e', 'started': 't', 'enabled': 't'}
+        self.server.manager(MGR_CMD_CREATE, QUEUE, a, id='workq2')
+
+        a = {'Resource_List.select': '1:ncpus=1', ATTR_q: 'workq2'}
+        jids1 = self.submit_jobs(3, user=TEST_USER, attrs=a)
+        jids2 = self.submit_jobs(3, user=TEST_USER2, attrs=a)
+
+        self.server.manager(MGR_CMD_SET, SERVER,
+                            {'scheduling': 'True'})
+
+        # Two equivalence classes.  One for the resource eating job and
+        # one for all jobs in workq2.
+        self.scheduler.log_match("Number of job equivalence classes: 2",
+                                 starttime=self.t)
```

Comment:
This test works as described, but I think you should take it one step further.  Submit another job to workq as a different user and check for 3 equiv classes.